### PR TITLE
[fix](truncate) fix bug that truncate partition throw NPE

### LINKED ISSUE #9340: [Bug] truncate partition with NPE
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Version

1.0.0

### What's Wrong?

```
CREATE TABLE tbl (
  `date_id` date NULL,
  `column2` tinyint(4) NULL
) ENGINE=OLAP
DUPLICATE KEY(`date_id`, `column2`)
PARTITION BY RANGE(`date_id`)
(
PARTITION p20200809 VALUES [('2020-08-09'), ('2020-08-10')),
PARTITION P20200810 VALUES [('2020-08-10'), ('2020-08-11')),
PARTITION P20200811 VALUES [('2020-08-11'), ('2020-08-12')),
PARTITION p20200812 VALUES [('2020-08-12'), ('2020-08-13')),
)
DISTRIBUTED BY HASH(`client_id`) BUCKETS 1
PROPERTIES (
"replication_allocation" = "tag.location.default: 1"
);

truncate table tbl partition(P20200809);
```

Error:
```
java.lang.NullPointerException: null
        at org.apache.doris.catalog.Catalog.truncateTable(Catalog.java:6782) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.DdlExecutor.execute(DdlExecutor.java:246) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.handleDdlStmt(StmtExecutor.java:1520) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.execute(StmtExecutor.java:432) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.StmtExecutor.execute(StmtExecutor.java:307) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.handleQuery(ConnectProcessor.java:212) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.dispatch(ConnectProcessor.java:349) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.qe.ConnectProcessor.processOnce(ConnectProcessor.java:538) ~[palo-fe.jar:1.0-SNAPSHOT]
        at org.apache.doris.mysql.nio.ReadListener.lambda$handleEvent$0(ReadListener.java:50) ~[palo-fe.jar:1.0-SNAPSHOT]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142) ~[?:1.8.0_131]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617) ~[?:1.8.0_131]
        at java.lang.Thread.run(Thread.java:748) ~[?:1.8.0_131]
```

### What You Expected?

truncate ok

### How to Reproduce?

This is because the real partition name is `p20200809`, but in truncate stmt,
it is `P20200809`.
The partition name is case insensitive, but we didn't handle it correctly in truncate operation.

### Anything Else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: close #9340

## Problem Summary:

1. partition name is case insensitive.
2. add a simple help-resource.zip to help pass the FE ut.

## Checklist(Required)

1. Does it affect the original behavior: (No)
3. Has unit tests been added: (Yes)
4. Has document been added or modified: (No Need)
5. Does it need to update dependencies: (No)
6. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java [hunk 1]
```diff
@@ -1286,11 +1286,13 @@ public OlapTable selectiveCopy(Collection<String> reservedPartitions, IndexExtSt
             return copied;
         }
 
-        Set<String> partNames = Sets.newTreeSet(String.CASE_INSENSITIVE_ORDER);
+        Set<String> partNames = Sets.newHashSet();
         partNames.addAll(copied.getPartitionNames());
 
+        // partition name is case insensitive:
+        Set<String> lowerReservedPartitionNames = reservedPartitions.stream().map(n -> n.toLowerCase()).collect(Collectors.toSet());
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1293]:
```
n -> n.toLowerCase()
```
suggest code
```
String::toLowerCase
```

**Review by EmmyMiao87 — COMMENTED** [review]
What is this used for help-resource.zip?

**Comment by morningman** [comment]
> What is this used for help-resource.zip?

In #9306, I add a strict check to verify the format of "help" document, but it results in that the FE ut can not start locally
duo to miss the help-resource.zip.
So I add one for the FE unit test. There is only one empty doc inside it, just for passing the check.

File: fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java (hunk 1 above)
Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/catalog/OlapTable.java:1293]: done

**Review by hf200012 — APPROVED** [review]
LGTM

**Pull request merged**